ls1intum/Artemis
# [Exam] Variants: Data model

### Checklist

### Motivation and Context

In order to be able to use Artemis for conducting exams, some adjustments to the data model are required.

### Description

- [x] We've implemented all new entities
- [x] We've added DTO classes for all new entities
- [x] We've created the corresponding Liquibase change-log
- [x] We've added corresponding TypeScript classes in the client
- [x] We've clarified everything that has *to be discussed*
- [x] We've added `extends AbstractAuditingEntity` to all new entities so that `created_date`, `last_modified`, ... are added to the database rows

![Exam Mode](https://user-images.githubusercontent.com/15168372/84052944-87bf7b00-a9b1-11ea-949c-e894081c8bd1.png)



## Additional changes
If exams are available, they are now part of the REST Call `/courses/for-dashboard`. We optimized the performance here, so please double check that we did not introduce unwanted behavior. 


## Steps for Testing
1. Login
2. Open the course overview, course details and exercises
3. Check that everything is displayed correctly (please compare this to the current state in `develop`
